### DISCUSSION TITLE
onCommand with portugues letters

### DISCUSSION THREAD

**Discussion by paulobreim** [DinoLeung/TeleDart#274] [Q&A]
When the user type /coleção teledart.onCommand does not recognize the word, work only with /colecao

Example:
  teledart
      .onCommand('coleção')
      .listen((message) => _checkmessage(teledart, message));  //never call

Is this a limitation of Telegram or can I handle it another way?

tks
paulo

**Comment by The0xArchitect** [comment]
Actually, I think if you cannot include any other than English alpha-numeric values in a bot command. That means, even if the user can send `/coleção` it is not matched as the `bot_command` entity.

Upon investigating the library code, I can see that the `onCommand` method only handles the messages with an `bot_command` entity. Which in-turn is the correct behavior. 

---
In other words, `/coleção` is not really considered as a command - hence handler attached with `onCommand` will not be executed for it. You may want to do:

```dart
teledart
.onMessage(keyword: '/coleção')
.listen((message) => _checkmessage(teledart, message)); 
```

This works, because now we're matching a simple text message.

**Comment by paulobreim** [comment]
ok, I understand and thank you.
To get around the problem I did the following, and it worked well.

```
  teledart
      .onMessage(keyword: '')
      .where(
        (message) => true,
      )
      .listen((message) async {
    if (message.text == '/coleção') {
      doSomething(teledart, message);
    }
  });
```
